analyzers/src/SonarAnalyzer.CSharp/Metrics/CSharpCyclomaticComplexityMetric.cs

```diff
@@ -201,12 +204,23 @@ public override void Visit(SyntaxNode node)
                     AddLocation(binaryPatternNode.OperatorToken);
                 }
 
-                base.Visit(node);
+                if (!IsStaticLocalFunction(node))
+                {
+                    base.Visit(node);
+                }
             }
 
             private void AddLocation(SyntaxToken node) => IncrementLocations.Add(new SecondaryLocation(node.GetLocation(), "+1"));
 
             private static bool HasBody(SyntaxNode node) => node.ChildNodes().AnyOfKind(SyntaxKind.Block);
+
+            private static bool IsStaticLocalFunction(GlobalStatementSyntax node) =>
+                node.ChildNodes().FirstOrDefault(y => y.IsKind(SyntaxKindEx.LocalFunctionStatement)) is { } localFunction
```

Comment:
Wouldn't be easier and safer to check the `node.Statement` if it's a local function?

`ChildNodes()` will iterate all children but the function should be the first one. I don't have a counterexample though :)